gravitee-apim-portal-webui-next/src/app/api-details/api-tab-documentation/api-tab-documentation.component.ts

```diff
@@ -15,31 +15,64 @@
  */
 import { AsyncPipe } from '@angular/common';
 import { Component, Input, OnInit } from '@angular/core';
+import { ActivatedRoute, Router, RouterModule } from '@angular/router';
+import { catchError, combineLatest, filter, map, Observable, of, switchMap } from 'rxjs';
 
+import { LoaderComponent } from '../../../components/loader/loader.component';
+import { PageComponent } from '../../../components/page/page.component';
 import { PageTreeComponent, PageTreeNode } from '../../../components/page-tree/page-tree.component';
 import { Page } from '../../../entities/page/page';
 import { PageService } from '../../../services/page.service';
 
+interface SelectedPageData {
+  result?: Page;
+  error?: unknown;
+}
+
 @Component({
   selector: 'app-api-tab-documentation',
   standalone: true,
-  imports: [PageTreeComponent, AsyncPipe],
+  imports: [PageTreeComponent, AsyncPipe, PageComponent, RouterModule, LoaderComponent],
   templateUrl: './api-tab-documentation.component.html',
   styleUrl: './api-tab-documentation.component.scss',
 })
 export class ApiTabDocumentationComponent implements OnInit {
   @Input()
-  pages: Page[] = [];
-  pageNodes: PageTreeNode[] = [];
-  selectedPage: string = '';
+  page!: string;
+  pageNodes$: Observable<PageTreeNode[]> = of([]);
+  selectedPageData$: Observable<SelectedPageData> = of();
 
-  constructor(private pageService: PageService) {}
+  private apiId$: Observable<string>;
+
+  constructor(
+    private pageService: PageService,
+    private router: Router,
+    private activatedRoute: ActivatedRoute,
+  ) {
+    this.apiId$ = this.activatedRoute.parent ? this.activatedRoute.parent.params.pipe(map(params => params['apiId'])) : of();
```

Comment:
Note for later, in case our app would grow and have many nested routes we can consider using `paramsInheritanceStrategy `

Reply:
Added!